blueprints/data-solutions/vertex-mlops/variables.tf

```diff
@@ -0,0 +1,228 @@
+  type        = string
+  default     = null
+}
+
+variable "kms_service_agents" {
+  description = "KMS IAM configuration in as service => [key]."
+  type        = map(list(string))
+  default     = {}
+}
+
+variable "labels" {
+  description = "Labels to be assigned at project level."
+  type        = map(string)
+  default     = {}
+}
+
+
+variable "notebooks" {
+  description = "Vertex AI workbenchs to be deployed."
+  type = map(object({
+    owner                 = string
+    region                = string
+    subnet                = string
+    internal_ip_only      = bool
+    idle_shutdown_timeout = bool
+  }))
+  default = null
+}
+
+variable "org_policies" {
+  description = "Org-policy overrides at project level."
+  type = map(object({
+    inherit_from_parent = optional(bool) # for list policies only.
+    reset               = optional(bool)
+
+    # default (unconditional) values
+    allow = optional(object({
+      all    = optional(bool)
+      values = optional(list(string))
+    }))
+    deny = optional(object({
+      all    = optional(bool)
+      values = optional(list(string))
+    }))
+    enforce = optional(bool, true) # for boolean policies only.
+
+    # conditional values
+    rules = optional(list(object({
+      allow = optional(object({
+        all    = optional(bool)
+        values = optional(list(string))
+      }))
+      deny = optional(object({
+        all    = optional(bool)
+        values = optional(list(string))
+      }))
+      enforce = optional(bool, true) # for boolean policies only.
+      condition = object({
+        description = optional(string)
+        expression  = optional(string)
+        location    = optional(string)
+        title       = optional(string)
+      })
+    })), [])
+  }))
+  default  = {}
+  nullable = false
+}
+
+variable "organization_domain" {
+  description = "Organization domain."
+  type        = string
+}
+
+variable "prefix" {
+  description = "Prefix used for the project id."
+  type        = string
+  default     = null
+}
+
+variable "project_create" {
+  description = "Create project. When set to false, uses a data source to reference existing project."
+  type        = bool
+  default     = true
+}
+
+variable "project_id" {
+  description = "Project id."
+  type        = string
+}
+variable "project_services" {
+  description = "List of core services enabled on all projects."
+  type        = list(string)
+  default = [
+    "aiplatform.googleapis.com",
+    "artifactregistry.googleapis.com",
+    "bigquery.googleapis.com",
+    "cloudbuild.googleapis.com",
+    "compute.googleapis.com",
+    "datacatalog.googleapis.com",
+    "dataflow.googleapis.com",
+    "iam.googleapis.com",
+    "monitoring.googleapis.com",
+    "notebooks.googleapis.com",
+    "secretmanager.googleapis.com",
+    "servicenetworking.googleapis.com",
+    "serviceusage.googleapis.com"
+  ]
+}
+
+variable "region" {
```

Comment:
We need a location (for GCS) and a region (for regional resources)

Reply:
In this example, I'm more leaned towards regional resources (included GCS(artifacts) and datasets)

Reply:
I would mention this in the README mentioning rationals.

Reply:
Added some rationale in the README.